### PR TITLE
Tri-state prop changes

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
Change these props to use tri-state values
  additional-paths [receive | send] [disable]
  soft-reconfiguration inbound

This is for CSCuw40269.
I originally implemented these properties as multiple boolean properties (e.g.: cap_add_paths_send & cap_add_paths_send_disable); Jie had the same problem with log_neighbor_changes but coded it as a tri-state instead (:enable, :disable, :inherit). The team voted for the tri-states so this is just converting my props over to do the same. On the plus side it's reduced a lot of puppet code.

I also changed the prop names from cap_add_paths_* to additional_paths_* to match what bgp_af used.

This change is dependent on similar changes in node_utils.
Beaker, rubocop are clean.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -513,17 +513,11 @@ Optional max-occurrences value for `allowas_in`. Valid values are an integer val
 ##### `as_override`
 `as-override`. Valid values are true, false, or 'default'.
 
-##### `cap_add_paths_receive`
-`capability additional-paths receive`. Valid values are true, false, or 'default'. Related: `cap_add_paths_receive_disable`.
+##### `additional_paths_receive`
+`capability additional-paths receive`. Valid values are `enable` for basic command enablement; `disable` for disabling the command at the neighbor_af level (it adds the `disable` keyword to the basic command); and `inherit` to remove the command at this level (the command value is inherited from a higher bgp layer).
```
Comment (glennmatthews) [inline README.md]: 'bgp' or 'BGP'?

File: lib/puppet/type/cisco_bgp_neighbor_af.rb [hunk 2]
```diff
@@ -284,28 +281,22 @@ def self.title_patterns
...
-
-  newproperty(:cap_add_paths_send) do
-    desc 'cap_add_paths_send state. ' \
-         "Valid values are true, false or 'default'"
-    newvalues(:true, :false, :default)
+  newproperty(:additional_paths_receive) do
+    desc 'additional_paths_receive state. ' \
+         "Valid values are 'enable' for basic command enablement; 'disable' " \
+         "for disabling the command at the neighbor_af level; and 'inherit' " \
+         'to remove the command at this level.'
```
Comment (glennmatthews) [inline lib/puppet/type/cisco_bgp_neighbor_af.rb]: 'remove the command at this level' is not super clear - I prefer the wording you have in the README.

File: tests/beaker_tests/bgpneighboraf/test_bgpneighboraf.rb [hunk 3]
```diff
@@ -113,10 +113,6 @@
   :manifest_props => "
     allowas_in                     => 'default',
     allowas_in_max                 => 'default',
-    cap_add_paths_receive          => 'default',
-    cap_add_paths_receive_disable  => 'default',
-    cap_add_paths_send             => 'default',
-    cap_add_paths_send_disable     => 'default',
```
Comment (glennmatthews) [inline tests/beaker_tests/bgpneighboraf/test_bgpneighboraf.rb]: Why aren't you adding the new tri-state properties to replace these removals?

File: README.md (hunk 1 above)
Comment (chrisvanheuveln) [inline README.md]: fixed

File: lib/puppet/type/cisco_bgp_neighbor_af.rb (hunk 2 above)
Comment (chrisvanheuveln) [inline lib/puppet/type/cisco_bgp_neighbor_af.rb]: k, fixed

**Comment by robert-w-gries** [comment]
Can you merge the Rubocop changes from develop and push again?

File: tests/beaker_tests/bgpneighboraf/test_bgpneighboraf.rb (hunk 3 above)
Comment (chrisvanheuveln) [inline tests/beaker_tests/bgpneighboraf/test_bgpneighboraf.rb]: This was discussed during one of our PL's; ie. some properties (tri-states) shouldn't support explicit default. If you recall I was overruled on this one. :-)  
If you feel strongly about these particular properties then I'll rework the code to support it.

File: lib/puppet/type/cisco_bgp_neighbor_af.rb (hunk 2 above)
Comment (hunner) [inline lib/puppet/type/cisco_bgp_neighbor_af.rb]: Hmm, I think the updates haven't been pushed as the PR diff is still showing `remove the command at this level`

**Comment by chrisvanheuveln** [comment]
I addressed all of the outstanding comments, plus made a few additional changes to satisfy the post-merge rubocop rules.

**Comment by chrisvanheuveln** [comment]
Hold on. Seemed to have lost my earlier changes.

**Comment by chrisvanheuveln** [comment]
ok, good to go. Please merge. Thx.

**Pull request merged**